### DISCUSSION TITLE
Option to follow redirect for rewrite

### DISCUSSION THREAD

**Discussion by kochis** [vercel/next.js#43442] [Help]
One of the options in [node-http-proxy](https://github.com/http-party/node-http-proxy) is the ability to `followRedirects`. We are currently using this to handle a re-write using [http-proxy-middleware](https://github.com/chimurai/http-proxy-middleware#readme_) with a custom express server.
```js
server.use('/documentation', createProxyMiddleware({
  target: 'https://radarlabs.github.io',
  changeOrigin: true,
  followRedirects: true,
}));
```

We're looking to remove the need for the custom express server, and moving this proxy call to `rewrites` is one of the last blocking items, but haven't been able to figure out a way to handle the redirect issue. We have the following re-write in place:
```js
rewrites: async () => {
  return [
    {
      source: '/documentation/:path*',
      destination: 'https://radarlabs.github.io/documentation/:path*',
    };
  ];
}
```
But since it's doing an initial redirect on load (due to a difference in trailing slashes), the page will end up on radarlabs.github.io instead of the next.js domain.

Is there any way to preserve this behavior? If not, could it be something possible to add (I could take a crack myself if pointed to the relevant parts of the codebase 😊)

Thanks! Again, this is the last piece that's sort of stopping us from moving to vercel. Thanks!

**Comment by flekmatik** [comment]
Actually I think that it would be better to remove/hide rewrites functionality without this feature. 

In many use cases we want to hide destination servers and it's not obvious that internal server hostname would be revealed if it for any reason returns a redirect.

Or at least mention this fact in documentation.
